connector/src/main/java/org/geysermc/connector/network/translators/bedrock/BedrockSetLocalPlayerAsInitializedTranslator.java

```diff
@@ -44,10 +46,20 @@ public void translate(SetLocalPlayerAsInitializedPacket packet, GeyserSession se
 
                 for (PlayerEntity entity : session.getEntityCache().getEntitiesByType(PlayerEntity.class)) {
                     if (!entity.isValid()) {
-                        SkinUtils.requestAndHandleSkinAndCape(entity, session, null);
+                        SkinManager.requestAndHandleSkinAndCape(entity, session, null);
                         entity.sendPlayer(session);
                     }
                 }
+
+                // Send Skulls
+                for (PlayerEntity entity : session.getSkullCache().values()) {
+                    entity.spawnEntity(session);
+
+                    SkinManager.requestAndHandleSkinAndCape(entity, session, (skinAndCape) -> session.getConnector().getGeneralThreadPool().schedule(() -> {
```

Comment:
1: This code is leftover from when chunks could load prior to spawning. It may make a return since Bund has a PR open for that.
2: Pretty sure the schedule specifically needs to be called after the player spawns.

Reply:
Then you can make two loops. I'm pretty sure that'll be better when you have a lot of skulls.